src/lib.rs

```diff
@@ -468,13 +471,19 @@ pub mod pallet {
 				let reward_info = RewardInfo {
 					total_reward: *reward,
 					claimed_reward: initial_payment,
+					contributed_relay_addresses: vec![relay_account.clone()],
 				};
```

Comment:
You create this in every case, but only use it if this is the first contribution benefitting this native account. You could create it in the `if`.

Reply:
I use it in the other if branch too (with some more changes).